src/store.cc

```diff
@@ -1374,7 +1374,9 @@ StoreEntry::negativeCache()
 #else
         expires = squid_curtime;
 #endif
-    EBIT_SET(flags, ENTRY_NEGCACHED);
+    debugs(20, 6, "expires = " << expires << ", curtime = " << squid_curtime);
+    if (expires > squid_curtime)
+        EBIT_SET(flags, ENTRY_NEGCACHED);
```

Comment:
I do not insist on this change, but I think it would be better to move the new debugs() inside the `if` statement to both reduce noise and to ease detection of `ENTRY_NEGCACHED` entries. Reporting the number of seconds left may also be useful. Here is a sketch:

```suggestion
    
    if (expires > squid_curtime) {
        EBIT_SET(flags, ENTRY_NEGCACHED);
        debugs(20, 6, "expires = " << expires << " +" << (expires-squid_curtime) << ' ' << *this);
    }
```